go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+	assert.Nil(t, err)
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, "ServedFrom(master): source_keyspace\nServedFrom(replica): source_keyspace\n", *clusterInstance)
+	checkBlacklistedTables(t, sourceMasterTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceReplicaTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceRdOnlyTablet1, sourceKeyspace, []string{"/moving/", "view1"})
+	checkBlacklistedTables(t, sourceRdOnlyTablet2, sourceKeyspace, []string{"/moving/", "view1"})
+
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedFrom", "destination_keyspace/0", "replica")
+	assert.Nil(t, err)
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, "ServedFrom(master): source_keyspace\n", *clusterInstance)
+	checkBlacklistedTables(t, sourceMasterTablet, sourceKeyspace, nil)
+	checkBlacklistedTables(t, sourceReplicaTablet, sourceKeyspace, []string{"/moving/", "view1"})
+	checkBlacklistedTables(t, sourceRdOnlyTablet1, sourceKeyspace, []string{"/moving/", "view1"})
+	checkBlacklistedTables(t, sourceRdOnlyTablet2, sourceKeyspace, []string{"/moving/", "view1"})
+	checkClientConnRedirectionExecuteKeyrange(ctx, t, gconn, destinationKeyspace, []topodata.TabletType{topodata.TabletType_MASTER}, []string{"moving1", "moving2"})
+
+	// Cancel should fail now
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("CancelResharding", "destination_keyspace/0")
+	assert.NotNil(t, err)
+
+	// then serve master from the destination shards
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("MigrateServedFrom", "destination_keyspace/0", "master")
+	checkSrvKeyspaceServedFrom(t, cellj, destinationKeyspace, "", *clusterInstance)
+	checkBlacklistedTables(t, sourceMasterTablet, sourceKeyspace, []string{"/moving/", "view1"})
+	checkBlacklistedTables(t, sourceReplicaTablet, sourceKeyspace, []string{"/moving/", "view1"})
+	checkBlacklistedTables(t, sourceRdOnlyTablet1, sourceKeyspace, []string{"/moving/", "view1"})
+	checkBlacklistedTables(t, sourceRdOnlyTablet2, sourceKeyspace, []string{"/moving/", "view1"})
+
+	// check the binlog player is gone now
+	_ = destinationMasterTablet.VttabletProcess.WaitForBinLogPlayerCount(0)
+
+	// check the stats are correct
+	checkStats(t)
+
+	// now remove the tables on the source shard. The blacklisted tables
+	// in the source shard won't match any table, make sure that works.
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("ApplySchema", "-sql=drop view view1", "source_keyspace")
+	assert.Nil(t, err)
+
+	for _, t := range []string{"moving1", "moving2"} {
+		_ = clusterInstance.VtctlclientProcess.ExecuteCommand("ApplySchema", "--sql=drop table "+t, "source_keyspace")
+	}
+	for _, t := range []cluster.Vttablet{sourceMasterTablet, sourceReplicaTablet, sourceRdOnlyTablet1, sourceRdOnlyTablet2} {
+		_ = clusterInstance.VtctlclientProcess.ExecuteCommand("ReloadSchema", t.Alias)
```

Comment:
catch the error,  and assert Nil for above 2 places.

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)